stl/inc/chrono

```diff
@@ -2209,6 +2221,2949 @@ namespace chrono {
+        } else {
+            // Convert to the last leap second before or equal to _Time.
+            const auto& _Last_leap = *--_It;
+            const utc_seconds _Utc_leap_second{_Last_leap.date().time_since_epoch() + _It->_Elapsed() - seconds{1}};
+#ifdef __cpp_lib_concepts
+            const auto _Leap_cmp = _Utc_leap_second <=> _Time_floor;
+#else // ^^^ __cpp_lib_concepts / TRANSITION, GH-395 workaround vvv
+            const auto _Leap_cmp = _Utc_leap_second > _Time_floor  ? strong_ordering::greater
+                                 : _Utc_leap_second == _Time_floor ? strong_ordering::equal
+                                                                   : strong_ordering::less;
+#endif // ^^^ workaround
+            if (_Tzdb._All_ls_positive && _STD is_gt(_Leap_cmp)) { // Case (1)
+                --_It;
+            }
+            return {_Last_leap._Positive() && _STD is_eq(_Leap_cmp), _It->_Elapsed()};
+        }
+    }
+
+    // CLASS utc_clock
+    class utc_clock {
+    public:
+        using rep                       = system_clock::rep;
+        using period                    = system_clock::period;
+        using duration                  = _CHRONO duration<rep, period>;
+        using time_point                = _CHRONO time_point<utc_clock>;
+        static constexpr bool is_steady = system_clock::is_steady;
+
+        _NODISCARD static time_point now() {
+            return from_sys(system_clock::now());
+        }
+
+        template <class _Duration>
+        _NODISCARD static sys_time<common_type_t<_Duration, seconds>> to_sys(const utc_time<_Duration>& _Utc_time) {
+            using _CommonType = common_type_t<_Duration, seconds>;
+            const auto _Lsi{_CHRONO get_leap_second_info(_Utc_time)};
+            _CommonType _Ticks;
+            if (_Lsi.is_leap_second) {
+                const auto _Leap_sec_minus_one = _CHRONO floor<seconds>(_Utc_time.time_since_epoch()) - _Lsi.elapsed;
+                if constexpr (is_integral_v<typename _Duration::rep>) {
+                    constexpr auto _Delta{seconds{1} - _CommonType{1}};
+                    _Ticks = _Leap_sec_minus_one + _Delta;
+                } else {
+                    const auto _Leap_sec_begin = _CHRONO ceil<_CommonType>(_Leap_sec_minus_one + seconds{1});
+                    _Ticks = _CommonType{_STD nextafter(_Leap_sec_begin.count(), typename _CommonType::rep{0})};
+                }
+            } else {
+                _Ticks = _Utc_time.time_since_epoch() - _Lsi.elapsed;
+            }
+            return sys_time<_CommonType>{_Ticks};
+        }
+
+        template <class _Duration>
+        _NODISCARD static utc_time<common_type_t<_Duration, seconds>> from_sys(const sys_time<_Duration>& _Sys_time) {
+            const auto& _Tzdb      = _CHRONO get_tzdb();
+            const auto& _Ls_vector = _Tzdb.leap_seconds;
+            auto _It = _STD upper_bound(_Ls_vector.begin(), _Ls_vector.end(), _CHRONO floor<seconds>(_Sys_time));

```

Comment:
Technically `cbegin()` would be proper but meh, it is already a const&

Reply:
Yeah, either way we're going to get a `const_iterator`. I can see arguments for both (why repeat `c` when `const` has already been applied as early as possible, versus if you always say `cbegin` by default, any remaining `begin`s stand out). I'm inclined to leave this as-is.